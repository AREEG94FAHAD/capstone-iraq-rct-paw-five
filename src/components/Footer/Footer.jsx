import React, { useState, useEffect } from 'react'

import { NavLink } from 'react-router-dom'
import { Container, Row, Col, Image } from 'react-bootstrap'
import { useSelector } from 'react-redux'
import { useTranslation } from 'react-i18next'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import {
  faMapMarkerAlt,
  faMobile,
  faEnvelope,
  faPowerOff,
} from '@fortawesome/free-solid-svg-icons'
import {
  faFacebookF,
  faTwitter,
  faInstagram,
  faYoutube,
} from '@fortawesome/free-brands-svg-icons'

import 'bootstrap/dist/css/bootstrap.min.css'
import logo from '../../images/logo.svg'
import {
  HOME_ROUTE,
  ADOPT_ROUTE,
  ABOUT_ROUTE,
  ADD_APET_ROUTE,
  RESOURCE_CAT_ROUTE,
  RESOURCE_DOG_ROUTE,
  LOGIN_ROUTE,
  CONTACT_US_ROUTE,
} from '../../routes'

export default function Footer() {
  const globaleLang = useSelector(state => state.langReducer)
  const { t } = useTranslation()
  const [dir, setDir] = useState('ltr')

  const changeDir = () => {
    if (
      localStorage.getItem('lang') !== 'en' &&
      localStorage.getItem('lang') !== null
    ) {
      setDir('rtl')
    } else {
      setDir('ltr')
    }
  }

  useEffect(() => {
    changeDir()
  }, [globaleLang])

  return (
    <div dir={dir} style={{ backgroundColor: '#F2F2F1' }}>
      <Container>
        <Row>
          <Col className="text-center pt-5" lg={4} md={6}>
            <Image src={logo} rounded width="64px" height="64px" />
            <p className="text-dark">{t('footer.pageDescription')}</p>
            <Container fluid>
              <FontAwesomeIcon
                icon={faFacebookF}
                className="text-danger ml-2 mr-2"
              />
              <FontAwesomeIcon
                icon={faTwitter}
                className="text-danger ml-2 mr-2"
              />
              <FontAwesomeIcon
                icon={faInstagram}
                className="text-danger ml-2 mr-2"
              />
              <FontAwesomeIcon
                icon={faYoutube}
                className="text-danger ml-2 mr-2"
              />
            </Container>
          </Col>
          <Col className="text-center" lg={4} md={6}>
            <h5 className="pt-5" align="center">
              {t('footer.explorLinks')}
            </h5>
            <Container fluid className="">
              <div className="list-unstyled row text-center">
                <li className="col-6 p-2 ">
                  <NavLink to={HOME_ROUTE} className="text-dark">
                    {t('navbar.home')}
                  </NavLink>
                </li>
                <li className="col-6 p-2">
                  <NavLink to={ABOUT_ROUTE} className="text-dark">
                    {t('navbar.adopt')}
                  </NavLink>
                </li>
                <li className="col-6 p-2">
                  <NavLink to={ADOPT_ROUTE} className="text-dark">
                    {t('navbar.about')}
                  </NavLink>
                </li>
                <li className="col-6 p-2">
                  <NavLink to={ADD_APET_ROUTE} className="text-dark">
                    {t('navbar.addapet')}
                  </NavLink>
                </li>
                <li className="col-6 p-2">
                  <NavLink to={RESOURCE_CAT_ROUTE} className="text-dark">
                    {t('navbar.resources.1')}
                  </NavLink>
                </li>
                <li className="col-6 p-2">
                  <NavLink to={RESOURCE_DOG_ROUTE} className="text-dark">
                    {t('navbar.resources.2')}
                  </NavLink>
                </li>
                <li className="col-6 p-2">
                  <NavLink to={CONTACT_US_ROUTE} className="text-dark">
                    {t('navbar.contact')}
                  </NavLink>
                </li>
                <li className="col-6 p-2">
                  <NavLink to={LOGIN_ROUTE} className="text-dark">
                    {t('navbar.logIn')}
                  </NavLink>
                </li>
              </div>
            </Container>
          </Col>
          <Col className="m-auto" lg={4} md={6}>
            <h5 className="pt-5" align="center">
              {t('footer.contactInfo')}
            </h5>
            <Container fluid className="text-center">
              <li className="m-3 list-unstyled text-dark">
                <FontAwesomeIcon
                  icon={faMapMarkerAlt}
                  className=" text-danger ml-2 mr-2"
                />
                {t('footer.cites')}{' '}
              </li>
              <li className="m-3 list-unstyled  text-dark">
                <FontAwesomeIcon
                  icon={faMobile}
                  className=" text-danger ml-2 mr-2"
                />

                {t('footer.phoneNo')}
              </li>
              <li className="m-3 list-unstyled text-dark">
                <FontAwesomeIcon
                  icon={faEnvelope}
                  className="text-danger  ml-2 mr-2"
                />
                {t('footer.email')}
              </li>
              <li className="m-3 list-unstyled text-dark">
                <FontAwesomeIcon
                  icon={faPowerOff}
                  className="text-danger ml-2 mr-2"
                />

                {t('footer.workTime')}
              </li>
            </Container>
          </Col>
        </Row>
      </Container>
      <Container className="theme-color text-center p-3 text-light" fluid>
        <h6>&#169; {t('footer.copyright')}</h6>
      </Container>
    </div>
  )
}
