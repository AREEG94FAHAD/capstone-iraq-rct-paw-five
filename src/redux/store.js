import { createStore, applyMiddleware } from 'redux'
import { composeWithDevTools } from 'redux-devtools-extension'
import logger from 'redux-logger'
import combineReducers from './reducers/AllReducers'

const store = createStore(
  combineReducers,
  composeWithDevTools(applyMiddleware(logger))
)

export default store
