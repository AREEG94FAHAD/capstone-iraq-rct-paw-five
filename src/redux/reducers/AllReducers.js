import { combineReducers } from 'redux'
import petReducer, {
  langReducer,
  myPets,
  deleteMyPet,
  updateMyPet,
} from './petReducer'
import user from './user'

export default combineReducers({
  langReducer,
  pets: petReducer,
  user,
  myPets,
  deleteMyPet,
  updateMyPet,
})
