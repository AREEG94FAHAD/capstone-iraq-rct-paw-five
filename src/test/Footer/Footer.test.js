import React from 'react'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import { createStore } from 'redux'
import { BrowserRouter as Router } from 'react-router-dom'
import Footer from '../../components/Footer/Footer'
import AllReducers from '../../redux/reducers/AllReducers'
const store = createStore(AllReducers)

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Router>
          <Footer store={store} />
        </Router>
      </Provider>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
