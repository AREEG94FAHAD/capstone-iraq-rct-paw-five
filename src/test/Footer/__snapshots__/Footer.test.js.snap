// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  dir="ltr"
  style={
    Object {
      "backgroundColor": "#F2F2F2",
    }
  }
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="text-center pt-5 col-lg-4 col-md-6"
      >
        <img
          className="rounded"
          height="64px"
          src="logo.svg"
          width="64px"
        />
        <p
          className="text-dark"
        >
          footer.pageDescription
        </p>
        <div
          className="container-fluid"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-facebook-f fa-w-10 text-danger ml-2 mr-2"
            data-icon="facebook-f"
            data-prefix="fab"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 320 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M279.14 288l14.22-92.66h-88.91v-60.13c0-25.35 12.42-50.06 52.24-50.06h40.42V6.26S260.43 0 225.36 0c-73.22 0-121.08 44.38-121.08 124.72v70.62H22.89V288h81.39v224h100.17V288z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-twitter fa-w-16 text-danger ml-2 mr-2"
            data-icon="twitter"
            data-prefix="fab"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-instagram fa-w-14 text-danger ml-2 mr-2"
            data-icon="instagram"
            data-prefix="fab"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M224.1 141c-63.6 0-114.9 51.3-114.9 114.9s51.3 114.9 114.9 114.9S339 319.5 339 255.9 287.7 141 224.1 141zm0 189.6c-41.1 0-74.7-33.5-74.7-74.7s33.5-74.7 74.7-74.7 74.7 33.5 74.7 74.7-33.6 74.7-74.7 74.7zm146.4-194.3c0 14.9-12 26.8-26.8 26.8-14.9 0-26.8-12-26.8-26.8s12-26.8 26.8-26.8 26.8 12 26.8 26.8zm76.1 27.2c-1.7-35.9-9.9-67.7-36.2-93.9-26.2-26.2-58-34.4-93.9-36.2-37-2.1-147.9-2.1-184.9 0-35.8 1.7-67.6 9.9-93.9 36.1s-34.4 58-36.2 93.9c-2.1 37-2.1 147.9 0 184.9 1.7 35.9 9.9 67.7 36.2 93.9s58 34.4 93.9 36.2c37 2.1 147.9 2.1 184.9 0 35.9-1.7 67.7-9.9 93.9-36.2 26.2-26.2 34.4-58 36.2-93.9 2.1-37 2.1-147.8 0-184.8zM398.8 388c-7.8 19.6-22.9 34.7-42.6 42.6-29.5 11.7-99.5 9-132.1 9s-102.7 2.6-132.1-9c-19.6-7.8-34.7-22.9-42.6-42.6-11.7-29.5-9-99.5-9-132.1s-2.6-102.7 9-132.1c7.8-19.6 22.9-34.7 42.6-42.6 29.5-11.7 99.5-9 132.1-9s102.7-2.6 132.1 9c19.6 7.8 34.7 22.9 42.6 42.6 11.7 29.5 9 99.5 9 132.1s2.7 102.7-9 132.1z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-youtube fa-w-18 text-danger ml-2 mr-2"
            data-icon="youtube"
            data-prefix="fab"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M549.655 124.083c-6.281-23.65-24.787-42.276-48.284-48.597C458.781 64 288 64 288 64S117.22 64 74.629 75.486c-23.497 6.322-42.003 24.947-48.284 48.597-11.412 42.867-11.412 132.305-11.412 132.305s0 89.438 11.412 132.305c6.281 23.65 24.787 41.5 48.284 47.821C117.22 448 288 448 288 448s170.78 0 213.371-11.486c23.497-6.321 42.003-24.171 48.284-47.821 11.412-42.867 11.412-132.305 11.412-132.305s0-89.438-11.412-132.305zm-317.51 213.508V175.185l142.739 81.205-142.739 81.201z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </div>
      </div>
      <div
        className="text-center col-lg-4 col-md-6"
      >
        <h5
          align="center"
          className="pt-5"
        >
          footer.explorLinks
        </h5>
        <div
          className="container-fluid"
        >
          <div
            className="list-unstyled row text-center"
          >
            <li
              className="col-6 p-2 "
            >
              <a
                aria-current="page"
                className="text-dark active"
                href="/"
                onClick={[Function]}
                style={Object {}}
              >
                navbar.home
              </a>
            </li>
            <li
              className="col-6 p-2"
            >
              <a
                aria-current={null}
                className="text-dark"
                href="/about"
                onClick={[Function]}
              >
                navbar.adopt
              </a>
            </li>
            <li
              className="col-6 p-2"
            >
              <a
                aria-current={null}
                className="text-dark"
                href="/adopt"
                onClick={[Function]}
              >
                navbar.about
              </a>
            </li>
            <li
              className="col-6 p-2"
            >
              <a
                aria-current={null}
                className="text-dark"
                href="/addpet"
                onClick={[Function]}
              >
                navbar.addapet
              </a>
            </li>
            <li
              className="col-6 p-2"
            >
              <a
                aria-current={null}
                className="text-dark"
                href="/cat"
                onClick={[Function]}
              >
                navbar.resources.1
              </a>
            </li>
            <li
              className="col-6 p-2"
            >
              <a
                aria-current={null}
                className="text-dark"
                href="/dog"
                onClick={[Function]}
              >
                navbar.resources.2
              </a>
            </li>
            <li
              className="col-6 p-2"
            >
              <a
                aria-current={null}
                className="text-dark"
                href="/contactus"
                onClick={[Function]}
              >
                navbar.contact
              </a>
            </li>
            <li
              className="col-6 p-2"
            >
              <a
                aria-current={null}
                className="text-dark"
                href="/login"
                onClick={[Function]}
              >
                navbar.logIn
              </a>
            </li>
          </div>
        </div>
      </div>
      <div
        className="m-auto col-lg-4 col-md-6"
      >
        <h5
          align="center"
          className="pt-5"
        >
          footer.contactInfo
        </h5>
        <div
          className="text-center container-fluid"
        >
          <li
            className="m-3 list-unstyled text-dark"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-map-marker-alt fa-w-12  text-danger ml-2 mr-2"
              data-icon="map-marker-alt"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 384 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M172.268 501.67C26.97 291.031 0 269.413 0 192 0 85.961 85.961 0 192 0s192 85.961 192 192c0 77.413-26.97 99.031-172.268 309.67-9.535 13.774-29.93 13.773-39.464 0zM192 272c44.183 0 80-35.817 80-80s-35.817-80-80-80-80 35.817-80 80 35.817 80 80 80z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
            footer.cites
             
          </li>
          <li
            className="m-3 list-unstyled  text-dark"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-mobile fa-w-10  text-danger ml-2 mr-2"
              data-icon="mobile"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 320 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M272 0H48C21.5 0 0 21.5 0 48v416c0 26.5 21.5 48 48 48h224c26.5 0 48-21.5 48-48V48c0-26.5-21.5-48-48-48zM160 480c-17.7 0-32-14.3-32-32s14.3-32 32-32 32 14.3 32 32-14.3 32-32 32z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
            footer.phoneNo
          </li>
          <li
            className="m-3 list-unstyled text-dark"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-envelope fa-w-16 text-danger  ml-2 mr-2"
              data-icon="envelope"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M502.3 190.8c3.9-3.1 9.7-.2 9.7 4.7V400c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V195.6c0-5 5.7-7.8 9.7-4.7 22.4 17.4 52.1 39.5 154.1 113.6 21.1 15.4 56.7 47.8 92.2 47.6 35.7.3 72-32.8 92.3-47.6 102-74.1 131.6-96.3 154-113.7zM256 320c23.2.4 56.6-29.2 73.4-41.4 132.7-96.3 142.8-104.7 173.4-128.7 5.8-4.5 9.2-11.5 9.2-18.9v-19c0-26.5-21.5-48-48-48H48C21.5 64 0 85.5 0 112v19c0 7.4 3.4 14.3 9.2 18.9 30.6 23.9 40.7 32.4 173.4 128.7 16.8 12.2 50.2 41.8 73.4 41.4z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
            footer.email
          </li>
          <li
            className="m-3 list-unstyled text-dark"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-power-off fa-w-16 text-danger ml-2 mr-2"
              data-icon="power-off"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M400 54.1c63 45 104 118.6 104 201.9 0 136.8-110.8 247.7-247.5 248C120 504.3 8.2 393 8 256.4 7.9 173.1 48.9 99.3 111.8 54.2c11.7-8.3 28-4.8 35 7.7L162.6 90c5.9 10.5 3.1 23.8-6.6 31-41.5 30.8-68 79.6-68 134.9-.1 92.3 74.5 168.1 168 168.1 91.6 0 168.6-74.2 168-169.1-.3-51.8-24.7-101.8-68.1-134-9.7-7.2-12.4-20.5-6.5-30.9l15.8-28.1c7-12.4 23.2-16.1 34.8-7.8zM296 264V24c0-13.3-10.7-24-24-24h-32c-13.3 0-24 10.7-24 24v240c0 13.3 10.7 24 24 24h32c13.3 0 24-10.7 24-24z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
            footer.workTime
          </li>
        </div>
      </div>
    </div>
  </div>
  <div
    className="theme-color text-center p-3 text-light container-fluid"
  >
    <h6>
      © 
      footer.copyright
    </h6>
  </div>
</div>
`;
